services/impl/src/main/java/org/exoplatform/automatic/translation/impl/connectors/GoogleTranslateConnector.java

```diff
@@ -16,12 +16,96 @@
  */
 package org.exoplatform.automatic.translation.impl.connectors;
 
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.DefaultConnectionReuseStrategy;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
 import org.exoplatform.automatic.translation.api.AutomaticTranslationComponentPlugin;
 import org.exoplatform.commons.api.settings.SettingService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
 
 public class GoogleTranslateConnector extends AutomaticTranslationComponentPlugin {
 
+  private static final Log    LOG                      = ExoLogger.getLogger(GoogleTranslateConnector.class);
+
+  private static final String GOOGLE_TRANSLATE_SERVICE = "google-translate";
+
+  private static final String API_URL                  = "https://translation.googleapis.com/language/translate/v2";
+
+  private static final String KEY_PARAM                = "key";
+
+  private static final String DATA_PATTERN             = "{'q': '{message}', 'target': '{targetLocale}'}";
+
+  private static final int DEFAULT_POOL_CONNECTION = 100;
+
+  private HttpClient httpClient;
+
   public GoogleTranslateConnector(SettingService settingService) {
     super(settingService);
+
+
+    PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+    connectionManager.setDefaultMaxPerRoute(DEFAULT_POOL_CONNECTION);
+    HttpClientBuilder httpClientBuilder = HttpClients.custom()
+                                                     .setConnectionManager(connectionManager)
+                                                     .setConnectionReuseStrategy(new DefaultConnectionReuseStrategy())
+                                                     .setMaxConnPerRoute(DEFAULT_POOL_CONNECTION);
+    this.httpClient = httpClientBuilder.build();
   }
+
+  @Override
+  public String translate(String message, Locale targetLocale) {
+    long startTime = System.currentTimeMillis();
+
+    String serviceUrl = API_URL + "?" + KEY_PARAM + "=" + getApiKey();
+    String data = DATA_PATTERN.replace("{message}", message).replace("{targetLocale}", targetLocale.getLanguage());
```

Comment:
in fact, the target lkocale provided comes from variable  eXo.env.portal.language='fr'. So if this variable have a variant, it will be use

Reply:
See https://github.com/Meeds-io/gatein-portal/blob/ce7ffb6d9d4f06b755e9ef0a121448745b45108c/web/portal/src/main/webapp/groovy/portal/webui/workspace/UIPortalApplication.gtmpl#L31-L35

Reply:
I'm not sure to understand your concern. 
I use the js variable eXo.env.portal.language when calling the rest service which make the translate
So, here, targetlocale take the value of exo.env.portal.language => if there is a variant, it will be used.

Reply:
`targetLocale.getLanguage()` will not include the country, for example (See the locale-config.xml file https://github.com/Meeds-io/gatein-portal/blob/develop/web/portal/src/main/webapp/WEB-INF/conf/common/locales-config.xml#L89-L101)

For both selected locales, the used language (if we don't include country/variant) will be `pt` only while it's different languages (`pt_BR` and `pt_PT`).

Reply:
ar you ok that the variant is present in exo.env.portal.language (and as I understand, it is), it will be used automatically for translation. It is already the case

Reply:
I changed targetLocale.getLocale to targetLocale.toLanguageTag() so that I have the variant is present in the Locale object